src/app-layout/__integ__/app-layout-tools.test.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { BasePageObject } from '@cloudscape-design/browser-test-tools/page-objects';
+import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
+
+import createWrapper from '../../../lib/components/test-utils/selectors';
+import { viewports } from './constants';
+
+const wrapper = createWrapper().findAppLayout();
+class AppLayoutToolsViewPage extends BasePageObject {
+  async openTools() {
+    await this.click(wrapper.findToolsToggle().toSelector());
+  }
+
+  async closeTools() {
+    await this.click(wrapper.findToolsClose().toSelector());
+  }
+}
+
+describe.each(['classic', 'refresh', 'refresh-toolbar'] as const)('%s', theme => {
+  function setupTest(testFn: (page: AppLayoutToolsViewPage) => Promise<void>, path: string) {
+    return useBrowser(async browser => {
+      const page = new AppLayoutToolsViewPage(browser);
+      const url = `#/light/app-layout/${path}`;
+      await page.setWindowSize(viewports.desktop);
+      const params = new URLSearchParams({
+        visualRefresh: `${theme.startsWith('refresh')}`,
+        appLayoutToolbar: `${theme === 'refresh-toolbar'}`,
+      });
+      await browser.url(`${url}?${params.toString()}`);
+      await page.waitForVisible(wrapper.findContentRegion().toSelector());
+      await testFn(page);
+    });
+  }
+
+  test(
+    'tools is displayed properly with split panel',
```

Comment:
If this was a regression test to prevent: https://github.com/cloudscape-design/components/pull/2826, I do not think it will help.

Tools element was hidden, but still caused the content overflow.

To catch that regression we need a test like this: https://github.com/cloudscape-design/components/blob/5b0e892123f96f8004d146d11251dee27894d481/src/app-layout/__integ__/mobile.test.ts#L52-L53

Or, actually extend that existing test to cover desktop sizes too.

---

P.S. also there is nothing app-layout-specific in that issue. Should we have a global check that none of our dev pages have horizontal scroll?

Reply:
nice testing method. I haven't came across that. I looked at the html and body with but they remained the same. I do like the horizontal scroll. That is how the regression picked it up